src/utils/xmlPartsUtils.ts

```diff
@@ -27,15 +28,19 @@ const updateWorkbookDataAndConfigurations = async (zip: JSZip, fileConfigs?: Fil
     await tableUtils.updateTableInitialDataIfNeeded(zip, tableData, updateQueryTable);
 };
 
-const updateWorkbookPowerQueryDocument = async (zip: JSZip, queryName: string, queryMashupDoc: string): Promise<void> => {
+const updateWorkbookPowerQueryDocument = async (zip: JSZip, queryName: string, queryMashupDoc: string, connectionOnlyQueryNames?: string[]): Promise<void> => {
     const old_base64: string | undefined = await pqUtils.getBase64(zip);
 
     if (!old_base64) {
         throw new Error(base64NotFoundErr);
     }
 
     const new_base64: string = await replaceSingleQuery(old_base64, queryName, queryMashupDoc);
-    await pqUtils.setBase64(zip, new_base64);
+    let updated_base64: string = new_base64;
+    if (connectionOnlyQueryNames && (connectionOnlyQueryNames.length > 0)) {
```

Comment:
You can clean a bit the code by encapsulating the logic which check if you need to add connection only queries inside the method: addConnectionOnlyQueriesIdNeeded => if (!connectionOnlyQueryNames || (connectionOnlyQueryNames.length == 0)) return base64